controller/src/scheduler.rs

```diff
@@ -0,0 +1,107 @@
+use cron::Schedule;
+use snafu::ResultExt;
+use std::env;
+use std::str::FromStr;
+
+// Defines the cron expression scheduler related env variable names
+const SCHEDULER_CRON_SECONDS_ENV_VAR: &str = "SCHEDULER_CRON_SECONDS";
+const SCHEDULER_CRON_MINUTES_ENV_VAR: &str = "SCHEDULER_CRON_MINUTES";
+const SCHEDULER_CRON_HOURS_ENV_VAR: &str = "SCHEDULER_CRON_HOURS";
+const SCHEDULER_CRON_DAY_OF_MONTH_ENV_VAR: &str = "SCHEDULER_CRON_DAY_OF_MONTH";
+const SCHEDULER_CRON_MONTH_ENV_VAR: &str = "SCHEDULER_CRON_MONTH";
+const SCHEDULER_CRON_DAY_OF_WEEK_ENV_VAR: &str = "SCHEDULER_CRON_DAY_OF_WEEK";
+const SCHEDULER_CRON_YEARS_ENV_VAR: &str = "SCHEDULER_CRON_YEARS";
+
+/// The module-wide result type.
+type Result<T> = std::result::Result<T, scheduler_error::Error>;
+
+struct BrupopCronScheduler {
+    seconds: String,
+    minutes: String,
+    hours: String,
+    day_of_month: String,
+    month: String,
+    day_of_week: String,
+    year: String,
+}
+
+impl BrupopCronScheduler {
+    pub fn new() -> Self {
+        let scheduler_cron_seconds = read_env_var(SCHEDULER_CRON_SECONDS_ENV_VAR).unwrap();
```

Comment:
Should probably get an `ERROR` level log message here instead of an unwrap. Can we change this to return `Result<Self>`?